addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -541,21 +542,10 @@ return AbstractModel.extend({
                 model: filter.write_model,
                 method: 'search_read',
                 domain: [["user_id", "=", session.uid]],
-                fields: [filter.write_field],
+                fields: this._searchFiltersFields(filter),
             })
             .then(function (res) {
-                var records = _.map(res, function (record) {
-                    var _value = record[filter.write_field];
-                    var value = _.isArray(_value) ? _value[0] : _value;
-                    var f = _.find(filter.filters, function (f) {return f.value === value;});
-                    var formater = fieldUtils.format[_.contains(['many2many', 'one2many'], field.type) ? 'many2one' : field.type];
-                    return {
-                        'id': record.id,
-                        'value': value,
-                        'label': formater(_value, field),
-                        'active': !f || f.active,
-                    };
-                });
+                var records = self._findFilterRecords(filter, res);
```

Comment:
I think the current implementation of the "filter checkbox" thing lacks a bit of genericity. 
The whole calendar is implemented using a "write_model" and a "write_field".

We could very much add a new property, say "filter_checkbox_field" that would imply all that behavior.
This way, all other usages of the calendar could easily implement the same feature by simply adding a boolean field to their model and providing the property.

I feel like it would also ease the code in CRM by only adding a field and a one-liner in the view.
All the added code to handle the filtering and the sync will be here instead.

What do you think?

(PS: If there are QUnit test of the calendar view, would be nice to include the new field there).